themes/default/content/blog/mlops-huggingface-llm-aws-sagemaker-python/index.md

```diff
@@ -0,0 +1,131 @@
+meta_image: "meta.png"
+authors:
+    - kat-morgan
+tags:
+    - ai
+    - ml
+    - iac
+    - aws
+    - llm
+    - vllm
+    - mlops
+    - llama
+    - llama2
+    - python
+    - sagemaker
+    - huggingface
+---
+
+Welcome to another post in the Pulumi + Python MLOps #MLOpsChallenge series!
+
+In this short tutorial we will deploy a publicly available Huggingface model on Amazon AWS SageMaker, and test a prompt using a short python script.
+
+We will be deploying:
+
+* The [NousResearch/Llama-2-7b-chat-hf](https://huggingface.co/NousResearch/Llama-2-7b-chat-hf) model on [Hugging Face](https://huggingface.co/) 
+* [Amazon AWS IAM Roles](https://www.pulumi.com/registry/packages/aws/api-docs/iam/role/)
+* [Amazon AWS SageMaker Model Endpoint](https://www.pulumi.com/registry/packages/aws/api-docs/sagemaker/model/)
+* [Amazon AWS CloudWatch MetricAlarm](https://www.pulumi.com/registry/packages/aws/api-docs/cloudwatch/metricalarm/)
+
+[Pulumi templates](../how-to-create-and-share-a-pulumi-template) enhance Infrastructure as Code (IaC) development by providing a pre-written code framework designed to help you bootstrap new pulumi projects quickly.
+
+You can use the `sagemaker-aws-python` template as a working python starting point for your own AWS SageMaker deployments, and customize the model you use, SageMaker configuration, and CloudWatch integration just to scratch the surface. Your only limits are your imagination.
+
+## Requirements
+
+* [Python3 (3.9+)](https://www.python.org/downloads/)
+* [Pulumi CLI](https://www.pulumi.com/docs/install/)
+* [Pulumi account](https://app.pulumi.com/signup)
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)
+* [Configured & Logged in with AWS Credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-authentication.html)
+
+## Instructions
+
+### 1. Login to Pulumi Cloud and initialize stack
+
+Let's begin by logging into Pulumi Cloud
+
+```bash
+# There are many ways to store Pulumi state, here we use Pulumi Cloud
+# Other state backends include S3, local file, and more
+pulumi login
+```
+
+### 2. Prepare a new Pulumi Project
+
+Here we create our new pulumi project directory and populate it from the `sagemaker-aws-python` [Pulumi template](https://github.com/pulumi/templates)
+
+```bash
+# Create a new directory & change directories into it
+mkdir newSagemaker && cd newSagemaker
+
+# Start your project from the sagemaker-aws-python template
+# Follow along with the prompts to create your new project and initialize a stack
+pulumi new sagemaker-aws-python
+```
+
+### 3. Deploy your model as a new SageMaker endpoint
+
+This step may take between 10 and 20 minutes while Amazon AWS builds your infrastructure. You can follow along in the console as resources are provisioned, or open the link displayed in terminal for your stack in a browser to follow along there too!
+
+```bash
+pulumi up
+```
```

Comment:
can we show a quick image or just text of what the pulumi up output looks like?

Reply:
+1. That was my first reaction after the reading the post. So what? I think a lot (most) of the code you are showing can be in the readme. Instead, show the interesting infrastructure bits and the resources, the output, what/how was created. Instead of all the details where and how to login

Reply:
done